test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,129 @@
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// hasStatus returns true of the given crdItem has a Status defined.
+func hasStatus(crdItem apiextensionsv1.CustomResourceDefinition) bool {
+	crdStatus := crdItem.Status
+	if len(crdStatus.Conditions) == 0 && len(crdStatus.StoredVersions) == 0 && crdStatus.AcceptedNames.Singular == "" {
+		return false
+	}
+	return true
+}
+
+// checkCRDs returns a list of names of CRDs that don't have subresource.status
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []apiextensionsv1.CustomResourceDefinition) []string {
+
+	// These 17 CRDs that, at the time this test was written, do not have subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	//
+	exceptionsList := sets.NewString(
+		"clusternetworks.network.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"nodes.config.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
+		"securitycontextconstraints.security.openshift.io",
+		"tuneds.tuned.openshift.io",
+	)
+
+	failures := []string{}
+	for _, crdItem := range crdItemList {
+
+		// This test is interested only in CRDs that end with "openshift.io".
+		if !strings.HasSuffix(crdItem.ObjectMeta.Name, "openshift.io") {
+			continue
+		}
+
+		crdName := crdItem.ObjectMeta.Name
+
+		// Skip CRDs in the exceptions list for now.
+		if exceptionsList.Has(crdName) {
+			continue
+		}
+
+		// Iterate through all versions of the Spec.
+		var versionName string
+
+		if hasStatus(crdItem) {
+			for i := 0; i < len(crdItem.Spec.Versions); i++ {
+
+				versionName = crdItem.Spec.Versions[i].Name
+				if crdItem.Spec.Versions[i].Subresources == nil {
+					failures = append(failures, fmt.Sprintf("CRD %s version %s missing subresources", crdName, versionName))
+				} else {
+					if crdItem.Spec.Versions[i].Subresources.Status == nil {
+						failures = append(failures, fmt.Sprintf("CRD %s version %s missing subresources.status", crdName, versionName))
+						// We will pass if one of them has a subresource.status.
+						break
+					}
+				}
+			}
+		} else {
+			failures = append(failures, fmt.Sprintf("CRD %s has no Status defined", crdName))
+		}
```

Comment:
This still looks like it's failing if any version is missing, David's comment awhile back suggested only to fail if NO versions had a status. In this case you'd want a bool for found and failing at the end if it was still false.

Reply:
ah, fixed.